util/settings/settings.go

```diff
@@ -649,6 +658,25 @@ func (mgr *SettingsManager) GetConfigManagementPlugins() ([]v1alpha1.ConfigManag
 	return plugins, nil
 }
 
+func (mgr *SettingsManager) GetEnableManifestGenerationForSourceType() (map[string]bool, error) {
```

Comment:
Is there any reason not to return `map[ApplicationSourceType]bool` instead? Might improve readability.

Reply:
gRPC does not support enums and force to use strings. Right now this method is only used to immediately pass settings to grpc request. So it is more convenient to return type that is already in consumable for grpc call.